ext/drreg/drreg.c

```diff
@@ -410,7 +410,12 @@ drreg_insert_restore_all(void *drcontext, instrlist_t *bb, instr_t *inst,
     uint aflags = (uint)(ptr_uint_t)drvector_get_entry(&pt->aflags.live, pt->live_idx);
     if (!pt->aflags.native &&
         (force_restore ||
-         TESTANY(EFLAGS_READ_ARITH, instr_get_eflags(inst, DR_QUERY_DEFAULT)) ||
+         /* On AArch64, unrecognized instructions are decoded as instances of a generic
+          * instruction, OP_xx. For these instructions conservatively restore aflags in
+          * case they read/write the aflags.
+          */
+         IF_AARCH64(instr_get_opcode(inst) == OP_xx ||)
+             TESTANY(EFLAGS_READ_ARITH, instr_get_eflags(inst, DR_QUERY_DEFAULT)) ||
```

Comment:
This is only handling reading flags: there are also points where drreg needs to update a spilled aflags value after an app instruction *writes* to the flags.  That point would need a workaround as well.

Reply:
I thought that this workaround handles writing flags as well. My understanding is that even if an unrecognized application instruction (`xx`) writes to the flags, this workaround will force the flags to be restored before `xx`, effectively preventing insertion of a restore after `xx` that overwrites the write to the flags by `xx`. If there is another save/restore pair by drreg after the 'xx' inst, it will not change the application view of the flags. Are you thinking of something else or am I missing something?

Reply:
Restoring is not unreserving.  drreg will assume that the app's flags value is still stored in TLS after an app read but not write of the flags, which will be incorrect after `xx` writes to the flags and changes that value.  drreg will proceed to use the wrong value when restoring on future reads or unreserves.

Reply:
I see.